libcontainer/cgroups/fscommon/utils.go

```diff
@@ -143,3 +143,26 @@ func GetCgroupParamString(path, file string) (string, error) {
 
 	return strings.TrimSpace(contents), nil
 }
+
+// ReadKeyValueFile reads all key-value pairs from the specified cgroup file,
+// returns a map from key to value.
+func ParseKeyValueFile(dir, file string) (map[string]uint64, error) {
+	content, err := cgroups.ReadFile(dir, file)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(content, "\n")
+	vals := make(map[string]uint64, len(lines))
+	for _, line := range lines {
+		arr := strings.Split(line, " ")
+		if len(arr) == 2 {
+			val, err := ParseUint(arr[1], 10, 64)
+			if err == nil {
+				vals[arr[0]] = val
+			}
+		}
+	}
+
+	return vals, nil
+}
```

Comment:
I don't like parsing the file to a map, and then looking up map values.

It would be better to use per-key switch statement. For example, see the implementation of `statCpu` in https://github.com/opencontainers/runc/blob/35988abe20851979d53d4a8f790b4c1b8800a10d/libcontainer/cgroups/fs2/cpu.go#L80